bouncerscript/utils.py

```diff
@@ -31,45 +32,43 @@ def load_json(path):
                              **retry_async_kwargs)
 
 
-async def _do_api_call(context, route, data, method='GET'):
-    """TODO"""
+async def _do_api_call(context, route, data, method='GET', session=None):
+    """Effective function doing the API call to the bouncer API endpoint"""
+    session = session or context.session
     bouncer_config = context.config["bouncer_config"][context.server]
     credentials = (bouncer_config["username"],
                    bouncer_config["password"])
     api_root = bouncer_config["api_root"]
     api_url = "%s/%s" % (api_root, route)
+    auth = aiohttp.BasicAuth(*credentials)
 
     kwargs = {'timeout': 60}
-    auth = None
     if data:
         kwargs['data'] = data
         method = 'POST'
-    if credentials:
-        auth = aiohttp.BasicAuth(*credentials)
-    async with aiohttp.ClientSession(auth=auth) as session:
-        log.info("Submitting to %s" % api_url)
-        try:
-            log.info("Performing a {} request to {} with kwargs {}".format(method,
-                                                                           api_url,
-                                                                           kwargs))
-            async with session.request(method, api_url, **kwargs) as resp:
-                log.info("Server response")
-                result = await resp.text()
-                log.info(result)
-                return result
-        except aiohttp.ClientError as e:
-            log.warning("Cannot access %s" % api_url)
-            traceback.print_exc(file=sys.stdout)
-            log.warning("Returned page source:")
-            log.warning(e.read())
-            raise
-        except aiohttp.ServerTimeoutError as e:
-            log.warning("Timed out accessing %s: %s" % (api_url, e))
-            raise
+
+    log.info("Submitting to %s" % api_url)
+    try:
+        log.info("Performing a {} request to {} with kwargs {}".format(method,
+                                                                       api_url,
+                                                                       kwargs))
+        async with session.request(method, api_url, auth=auth, **kwargs) as resp:
+            log.info("Server response")
+            result = await resp.text()
+            log.info(result)
+            return result
+    except aiohttp.ServerTimeoutError as e:
+        log.warning("Timed out accessing %s: %s" % (api_url, e))
+        raise
+    except aiohttp.ClientError as e:
+        log.warning("Cannot access %s" % api_url)
+        traceback.print_exc(file=sys.stdout)
+        raise
 
 
 async def product_exists(context, product_name):
```

Comment:
Nit: We could make clearer that the function returns a boolean, by using an interrogative form. For instance: `does_product_exist(context, product_name)`